src/charset.c

```diff
@@ -798,6 +798,40 @@ linetabsize(win_T *wp, linenr_T lnum)
 		       ml_get_buf(wp->w_buffer, lnum, FALSE), (colnr_T)MAXCOL);
 }
 
+/*
+ * Like linetabsize(), but excludes 'above'/'after'/'right'/'below' aligned
+ * virtual text, while keeping inline virtual text.
+ */
+    int
+linetabsize_no_outer(win_T *wp, linenr_T lnum)
+{
+#ifndef FEAT_PROP_POPUP
+    return linetabsize(wp, lnum);
+#else
+    chartabsize_T cts;
+    char_u *line = ml_get_buf(wp->w_buffer, lnum, FALSE);
+
+    init_chartabsize_arg(&cts, wp, lnum, 0, line, line);
+
+    int write_idx = 0;
+    for (int read_idx = 0; read_idx < cts.cts_text_prop_count; read_idx++)
+    {
+	textprop_T *tp = &cts.cts_text_props[read_idx];
+	if (tp->tp_col != MAXCOL)
+	{
+	    if (read_idx != write_idx)
+		cts.cts_text_props[write_idx] = *tp;
```

Comment:
I think you are leaking memory here.

Reply:
Oh, not here, but if `cts.cts_text_prop_count` gets changed to `0` through this filtering, then `cts.cts_text_props` won't be freed by `clear_chartabsize_arg`.

I've pushed a fixup commit to fix the memory leak.  Would you like me to rebase and squash the fixup?

Reply:
if you could squash, that would be prefered, thanks!